hatch_zipped_directory/builder.py

```diff
@@ -24,26 +26,43 @@
 
 __all__ = ["ZippedDirectoryBuilder"]
 
+ZipTime = Tuple[int, int, int, int, int, int]
+
 
 class ZipArchive:
-    def __init__(self, zipfd: ZipFile, root_path: str):
+    def __init__(self, zipfd: ZipFile, root_path: str, *, reproducible: bool):
```

Comment:
I made it keyword-only and explicit to be https://beta.ruff.rs/docs/rules/boolean-type-hint-positional-argument/ compatible

Reply:
I have no issue with making it keyword-only.  I would favor adding a default value, i.e.

```py
class ZipArchive:
    def __init__(self, zipfd: ZipFile, root_path: str, *, reproducible: bool = False):
        ...
```


This is admittedly a picky stylistic nit.

It is true that our code always provides a value for the argument, so it's not critical that we provide a default.  But I do view the `reproducible` option as "optional" behavior that is not central to the operation of `ZipArchive`.  Adding a default value expresses that notion. (It also prevents breaking any external code — though I suspect there is none — that may use `ZipArchive` directly.)

I don't have any strong preference as to whether the default should be `reproducible=False` or `reproducible=True`.  Defaulting to `False` maintains existing behavior, but, since hatch defaults to `reproducible=True`, that would be a reasonable default as well.